contracts/protocols/mainnet/erc4626/main.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+import "./interfaces.sol";
+
+contract Resolver {
+    struct VaultData {
+        bool isToken;
+        string name;
+        string symbol;
+        uint256 decimals;
+        address asset;
+        uint256 totalAssets;
+    }
+
+    function getVaultDetails(address vaultAddress) public view returns (VaultData memory) {
+        VaultData memory _vaultData;
+        VaultInterface vaultToken = VaultInterface(vaultAddress);
+        bool isToken = true;
+
+        try vaultToken.symbol() {} catch {
+            isToken = false;
+        }
+
+        try vaultToken.name() {} catch {
+            isToken = false;
+        }
+
+        try vaultToken.decimals() {} catch {
+            isToken = false;
+        }
+
```

Comment:
if `isToken` is false, return here. Don't proceed further.

Reply:
Also, validate if it has exchangePrice, asset, etc as well.

Reply:
I think we can only validate vaulToken.asset() like this:
            
try vaultToken.asset() {} catch {
    isToken = false;
    continue;
}

And what about the validating exchangePrice?